ibis/backends/pyspark/compiler.py

```diff
@@ -1,2137 +1,502 @@
-        result_table = result_table.drop(*col_to_drop)
-
-    if op.sort_keys:
-        sort_cols = [
-            t.translate(key, scope=scope, timecontext=timecontext, **kwargs)
-            for key in op.sort_keys
-        ]
-        result_table = result_table.sort(*sort_cols)
-
-    return filter_by_time_context(result_table, timecontext, adjusted_timecontext)
-
-
-@compiles(ops.SortKey)
-def compile_sort_desc(t, op, **kwargs):
-    col = t.translate(op.expr, **kwargs)
-    return col.asc() if op.ascending else col.desc()
-
-
-def compile_nan_as_null(compile_func):
-    @functools.wraps(compile_func)
-    def wrapper(t, op, *args, **kwargs):
-        compiled = compile_func(t, op, *args, **kwargs)
-        if options.pyspark.treat_nan_as_null and isinstance(op.dtype, dt.Floating):
-            return F.nanvl(compiled, F.lit(None))
+    @visit_node.register(ops.Field)
+    def visit_Field(self, op, *, rel, name):
+        result = super().visit_Field(op, rel=rel, name=name)
+        if op.dtype.is_floating() and options.pyspark.treat_nan_as_null:
+            return self.f.nanvl(result, sge.NULL)
         else:
-            return compiled
-
-    return wrapper
-
-
-@compiles(ops.TableColumn)
-@compile_nan_as_null
-def compile_column(t, op, **kwargs):
-    name = op.name
-    table = op.table
-    try:
-        name = f"`{table.name}`.`{name}`"
-    except AttributeError:
-        spark_df = t.translate(table, **kwargs)
-        return spark_df[name]
-    else:
-        return F.col(name)
-
-
-@compiles(ops.StructField)
-def compile_struct_field(t, op, **kwargs):
-    arg = t.translate(op.arg, **kwargs)
-    return arg[op.field]
-
+            return result
 
-@compiles(ops.StructColumn)
-def compile_struct_column(t, op, **kwargs):
-    return F.struct(
-        *(
-            t.translate(col, **kwargs).alias(name)
-            for name, col in zip(op.names, op.values)
+    @visit_node.register(ops.Cast)
+    def visit_Cast(self, op, *, arg, to):
+        if to.is_json():
+            if op.arg.dtype.is_string():
+                return arg
+            else:
+                return self.f.to_json(arg)
+        else:
+            return self.cast(arg, to)
+
+    @visit_node.register(ops.IsNull)
+    def visit_IsNull(self, op, *, arg):
+        is_null = arg.is_(sge.NULL)
+        is_nan = self.f.isnan(arg)
+        if op.arg.dtype.is_floating():
+            return sg.or_(is_null, is_nan)
+        else:
+            return is_null
+
+    @visit_node.register(ops.NotNull)
+    def visit_NotNull(self, op, *, arg):
+        is_not_null = arg.is_(sg.not_(sge.NULL))
+        is_not_nan = sg.not_(self.f.isnan(arg))
+        if op.arg.dtype.is_floating():
+            return sg.and_(is_not_null, is_not_nan)
+        else:
+            return is_not_null
+
+    @visit_node.register(ops.IsInf)
+    def visit_IsInf(self, op, *, arg):
+        if op.arg.dtype.is_floating():
+            return sg.or_(arg == self.POS_INF, arg == self.NEG_INF)
```

Comment:
I thought there was an `is_inf` function, but I could be wrong.

Reply:
Spark doesn't have that.